src/ape/api/providers.py

```diff
@@ -133,6 +135,22 @@ def decode(cls, data: dict) -> "ReceiptAPI":
         ...
 
 
+@abstractdataclass
+class BlockAPI:
+    gas_fee_data: Dict
+    consensus_data: Dict
```

Comment:
I like these, wonder if we could make them more explicit types, and offer relevant data from them e.g. `average_fee`, `total_gas_used`, `average_gas_used` and other relevant factors.

This is actually getting me thinking more about querying data (e.g. a data model for block that allows computing data over many blocks)

Reply:
```py
@abstractdataclass
class BlockGasDataAPI:
    @property
    @abstractmethod
    def average_fee(self) -> int:
        ...
```

and the like

Reply:
I like the easier querying this would make